commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -18,23 +19,75 @@
 
 
 import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
 
 public class MwVolleyApi {
 
     private static RequestQueue REQUEST_QUEUE;
     private static final Gson GSON = new GsonBuilder().create();
     private Context context;
+    private static String coordsLog;
+
+    protected static HashSet<String> categorySet;
+
+    //To check later on whether any nearby categories were found
+    public static boolean gpsCatExists;
 
     public MwVolleyApi(Context context) {
         this.context = context;
+        categorySet = new HashSet<String>();
+    }
+
+    //To get the list of categories for display
+    public static ArrayList<String> getGpsCat() {
+        ArrayList<String> list = new ArrayList<String>(categorySet);
+        return list;
     }
-    public void request(String apiUrl) {
+
+
+    public void request(String coords) {
+
+        coordsLog = coords;
+        String apiUrl = buildUrl(coords);
+        Log.d("Image", "URL: " + apiUrl);
+
         JsonRequest<QueryResponse> request = new QueryRequest(apiUrl,
                 new LogResponseListener<QueryResponse>(), new LogResponseErrorListener());
```

Comment:
Is it appropriate to keep these LogResponseListeners? Maybe you'd want to require the client pass in a listener instead?